test/functional/tools/dbkey_output_action.xml

```diff
@@ -0,0 +1,24 @@
+<tool id="dbkey_output_action" name="dbkey_output_action" version="0.1.0">
+    <command>echo foo > $mapped_reads</command>
+    <inputs>
+        <param name="input" type="data" />
+        <param name="index" type="select" label="Using reference genome">
+          <options from_data_table="test_fasta_indexes">
+            <filter type="data_meta" ref="input" key="dbkey" column="1" />
+            <validator type="no_options" message="No reference genome is available for the build associated with the selected input dataset" />
+          </options>
+        </param>
+    </inputs>
+    <outputs>
+        <data format="txt" name="mapped_reads">
+            <actions>
+                <action type="metadata" name="dbkey">
+                    <option type="from_data_table" name="test_fasta_indexes" column="1" offset="0">
+                        <filter type="param_value" column="0" value="#" compare="startswith" keep="False"/>
+                        <filter type="param_value" ref="index" column="0"/>
+                    </option>
+                </action>
+            </actions>
+        </data>
+    </outputs>
```

Comment:
Without a `<tests>` section this is not really testing that metadata assignment works, is it?

See https://github.com/galaxyproject/galaxy/pull/2795/files#r74952233 for a test template.